### PR TITLE
Update camera for Android 11

### LINKED ISSUE #2503: Qfield not opening external camera app on OnePlus 8 Pro Android 11
Qfield master-dev (d7be25) does not open the external camera app on an OnePlus 8 Pro running Android 11. Project is working with Samsung Phones. Qfield 1.10.0 does open the external camera app using the same project on OnePlus 8 Pro. 

### Reproduction Steps:

1. Setup Pictures in QGis following https://qfield.org/docs/de/prepare/add-1-n-pictures.html
2. Transfer to Phone using Qfield Cloud.
3. Set to use native camera
4. Try to take a picture 

### Expected behavior:

Opening the native camera app.

### Observed behavior:

White screen 

### Screenshots and GIFs

Screenshot_20220209-213306

**QField version:** Qfield master-dev (d7be25)

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: Working with "old" Qfield 1.10.0
* Problem can be reliably reproduced, doesn't happen randomly: only on OnePlus 8 Pro Smartphone
* Problem happens with all files and projects, not only some files or projects: Happens with different project as well

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes #2503

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 68db42634638adb76a670ec0a60178d8cfa90596:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-5cc3e038be55c3f005d82110cd3af801a028f16f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-5cc3e038be55c3f005d82110cd3af801a028f16f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-5cc3e038be55c3f005d82110cd3af801a028f16f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-5cc3e038be55c3f005d82110cd3af801a028f16f-x86.apk)

File: platform/android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java [hunk 1]
```diff
@@ -58,24 +62,35 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                Uri photoURI = FileProvider.getUriForFile(
-                    this, "ch.opengis.qfield.fileprovider", pictureFile);
-
-                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoURI);
-                startActivityForResult(takePictureIntent, 172);
-            }
+        Context context = getApplication().getApplicationContext();
+        File path = context.getExternalFilesDir(null);
+        File pictureFile =
+            File.createTempFile(pictureTempFileName, ".jpg", path);
```
Comment (nirvn) [inline platform/android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:68]: You probably want to use the suffix here

Comment (m-kuhn) [reply platform/android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:68]: Is the ".jpg" not enough?

Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:68]: The way we had it before was relying on suffix, which I assumed meant a .png would tell the camera app to save as png. 

If that assumption is wrong, we can get rid of the extra suffix string altogether.

Comment (m-kuhn) [reply platform/android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:68]: `createTempFile` expects the suffix separately, so if that's needed we need to split it off

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 5fdec04a56266dea19609bfcd0039a7be8256e71:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-637cb76e4b89973c81aca92c493e19e9256261e5-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-637cb76e4b89973c81aca92c493e19e9256261e5-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-637cb76e4b89973c81aca92c493e19e9256261e5-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-637cb76e4b89973c81aca92c493e19e9256261e5-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0d30c3b99d3994b41c6e2e3c74034a4a80f1973b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-042e202678d2b37d3f7b8e9c721927d61b704925-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-042e202678d2b37d3f7b8e9c721927d61b704925-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-042e202678d2b37d3f7b8e9c721927d61b704925-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-042e202678d2b37d3f7b8e9c721927d61b704925-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a867c2601c19928b06d1fd0595bced5a17b4cb40:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-b0a5fca01c35c977f0c92fe78305220c7466baed-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-b0a5fca01c35c977f0c92fe78305220c7466baed-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-b0a5fca01c35c977f0c92fe78305220c7466baed-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2486-b0a5fca01c35c977f0c92fe78305220c7466baed-x86.apk)

**Review by nirvn — APPROVED** [review]

**Pull request merged**